pkg/artifactory/resource/security/resource_artifactory_group.go

```diff
@@ -151,6 +151,34 @@ func groupParams(s *schema.ResourceData) (Group, bool, error) {
 	return group, includeUsers, nil
 }
 
+func GetDataSourceGroupSchema() map[string]*schema.Schema {
+	var includeUsersSchema = map[string]*schema.Schema{
+		"include_users": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Default:     false,
+			Description: "Setting includeUsers to true will return the group with its associated user list attached.",
+		},
+	}
+
+	includeGroupSchema := util.MergeMaps(GroupSchema, includeUsersSchema)
+	delete(includeGroupSchema, "detach_all_users")
+	return includeGroupSchema
+}
+
+func DataSourceGroupRead(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	group := Group{}
+	name := d.Get("name").(string)
+	includeUsers := d.Get("include_users").(string)
+	m.(*resty.Client).R().SetResult(&group).SetQueryParam("includeUsers", includeUsers).Get(GroupsEndpoint + name)
+
```

Comment:
Re my previous comment on removing the 404 check, etc. I didn't realize this is the resource. I thought this func is the copy in the data source. Please revert these changes here, and apply them to the data source instead.